src/components/user-form/AddUserForm.js

```diff
@@ -2,102 +2,128 @@ import React, { Component, createRef } from 'react';
+
+        console.log(first_name, last_name, email, address);
+            onAddUser({ first_name, last_name, email, address })
+        // todo 1: использовать функцию onUserAdd для создания нового пользователя здесь
+        //  у этого объекта должны быть проперти first_name, last_name, address, email
+
+        // todo 3: зачистить форму, вызвав функцию onReset
+    };
+
+
+    // todo 3: добавить функцию onReset, которая будет зачищать поля формы
+    //  обратите внимание, что тут нет стейта и мы обращаемся к инпутам с помощью ref
+
+    focusInput = () => {
+        console.log('focused');
+        this.lastNameRef.current.focus()
+    };
+
+    render() {
+        return (
+            <form onSubmit={this.onSubmit}>
+                {!!this.state.warning && <div>{this.state.warning}</div>}
+                <div className="form-group">
+                    <label htmlFor="firstNameInput">First name:</label>
+                    <input
+                        ref={this.firstNameRef}
+                        type="text"
+                        className="form-control"
+                        id="firstNameInput"
+                        placeholder="Example input"
+                    />
+                </div>
+
+                <div className="form-group">
+                    <label htmlFor="lastNameInput">Last name:</label>
+                    <input
+                        ref={this.lastNameRef}
+                        type="text"
+                        className="form-control"
+                        id="lastNameInput"
+                        placeholder="Example input"
+                    />
+                </div>
+
+                <div className="form-group">
+                    <label htmlFor="lastNameInput">Email:</label>
+                    <input
+                        ref={this.emailNameRef}
+                        type="text"
+                        className="form-control"
+                        id="lastNameInput"
+                        placeholder="Example input"
+                    />
+                    <input
+                        ref={this.addressNameRef}
+                        type="text"
+                        className="form-control"
+                        id="addressNameInput"
+                        placeholder="Example Address"
+                    ></input>
+                </div>
+
+
+                {/*
         todo 1: добавить инпут для ввода адресса
               передать ему атрибут под название ref наш созданный в строке 8 ref
         */}
 
-        <button type='submit' className="btn btn-primary m-2">Add </button>
-        <button type='button' onClick={this.focusInput}>focus</button>
+                <button type='submit' className="btn btn-primary m-2">Add </button>
+                <button type='button' onClick={this.focusInput} >focus</button>
+                <button type='reset' className="btn btn-primary m-2" onClick={() => this.onReset} >Reset</button>
```

Comment:
I guess this syntax is not correct `onClick={() => this.onReset}`
You can use like this `onClick={this.onReset}` or `onClick={() => this.onReset()}`
but the first variant is better.
as I understand, currently form's reset works because you specified type of button as 'reset' which is actually interesting
try to test how behavior would work if:
1) you remove onClick from reset btn at all
2) you will put type="button" to reset btn